arionBackend/transformation/esper/event_type_retriever.py

```diff
@@ -36,3 +37,15 @@ def find_feeding_event_types(input_query):
 				for dictionary in value:
 					for result in EventTypeRetriever.find_feeding_event_types(dictionary):
 						yield result
+
+	@staticmethod
+	def find_feeding_event_types_for_pattern(pattern):
+		"""
+		Finds all event types that are used in the pattern. They are detected by their CamelCase notation.
+		:param pattern: the pattern to find the event types in.
+		:return: yields event types that are found, must be caught in a list.
+		"""
+
+		matches = regex_findall("\W+?(?P<EventType>([A-Z][a-z]+?)+?)\W", pattern)

```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove this "\", add another "\" to escape it, or make this a raw string. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.tommartensen.de/coding_rules#rule_key=python%3AS1717)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove this "\", add another "\" to escape it, or make this a raw string. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.tommartensen.de/coding_rules#rule_key=python%3AS1717)